test/e2e/prometheusadapter_test.go

```diff
@@ -24,8 +24,88 @@ import (
 	"github.com/pkg/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
+	apiservicesv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
 )
 
+func isAPIServiceAvailable(conditions []apiservicesv1.APIServiceCondition) bool {
+	for _, condition := range conditions {
+		if condition.Type == apiservicesv1.Available && condition.Status == apiservicesv1.ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+func TestMetricsAPIAvailability(t *testing.T) {
+	var lastErr error
+	err := wait.Poll(time.Second, 5*time.Minute, func() (bool, error) {
+		metricsService, err := f.APIServicesClient.ApiregistrationV1().APIServices().Get("v1beta1.metrics.k8s.io", metav1.GetOptions{})
+		lastErr = errors.Wrap(err, "getting metrics APIService failed")
+		if err != nil {
+			return false, nil
+		}
+		if !isAPIServiceAvailable(metricsService.Status.Conditions) {
+			lastErr = errors.New("v1beta1.metrics.k8s.io apiservice is not available")
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		if err == wait.ErrWaitTimeout && lastErr != nil {
+			err = lastErr
+		}
+		log.Fatal(err)
+	}
+}
+
+func TestNodeMetricsPresence(t *testing.T) {
+	var lastErr error
+	err := wait.Poll(time.Second, 5*time.Minute, func() (bool, error) {
+		nodeMetrics, err := f.MetricsClient.MetricsV1beta1().NodeMetricses().List(metav1.ListOptions{})
+		lastErr = errors.Wrap(err, "getting metrics list failed")
+		if err != nil {
+			return false, nil
+		}
+		for _, item := range nodeMetrics.Items {
+			if item.Usage.Cpu() == nil || item.Usage.Memory() == nil {
+				lastErr = errors.New("node cpu or memory metric not found")
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+	if err != nil {
+		if err == wait.ErrWaitTimeout && lastErr != nil {
+			err = lastErr
+		}
+		log.Fatal(err)
+	}
+}
+
+func TestPodMetricsPresence(t *testing.T) {
+	var lastErr error
+	err := wait.Poll(time.Second, 5*time.Minute, func() (bool, error) {
+		podMetrics, err := f.MetricsClient.MetricsV1beta1().PodMetricses(f.Ns).List(metav1.ListOptions{})
+		lastErr = errors.Wrap(err, "getting metrics list failed")
+		if err != nil {
+			return false, nil
+		}
+		for _, pod := range podMetrics.Items {
+			for _, item := range pod.Containers {
+				if item.Usage.Cpu() == nil || item.Usage.Memory() == nil {
+					lastErr = errors.New("container cpu or memory metric not found")
```

Comment:
this should also verify that we are in fact getting results for all nodes returned by the nodes API

Reply:
Added by first checking if both lists are equal length and later iterating over list of pods.